sqlstats: stats flush should honor stopper quiesce

dhartunian opened the pull request:
Previously, the sql stats flush async tasks would continue potentially
long blocks of work during stopper quiescence. This change helps them
cancel mid-flush when the stopper is quiescing for orderly shutdown.

Resolves: https://github.com/cockroachdb/cockroach/issues/133106
Epic: None

Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/135336)
<!-- Reviewable:end -->

dhartunian committed:
sqlstats: stats flush should honor stopper quiesce

Previously, the sql stats flush async tasks would continue potentially
long blocks of work during stopper quiescence. This change helps them
cancel mid-flush when the stopper is quiescing for orderly shutdown.

Resolves: #133106
Epic: None

Release note: None

angles-n-daemons reviewed (approved):
Looks good to me.

dhartunian commented:
bors r+
